Sources/Fuzzilli/FuzzIL/AbstractInterpreter.swift

```diff
@@ -198,39 +262,48 @@ public struct AbstractInterpreter {
     
     /// Attempts to infer the return value type of the given function.
     private func inferCallResultType(of function: Variable) -> Type {
-        if let signature = type(of: function).functionSignature {
+        if let signature = state.type(of: function).functionSignature {
             return signature.outputType
         }
         
         return .unknown
     }
     
-    /// Sets the type of the given variable in the current state.
-    /// Return changed variable with its new type
     public mutating func setType(of v: Variable, to t: Type) {
-        stack[stack.count - 1][v] = t
+        state.setType(of: v, to: t)
```

Comment:
Since this is now an single-line function, maybe we can remove it?

Reply:
We need to keep this function to expose this functionality for `ProgramBuilder` (setting runtime types)